docs/fundamentals/persist-data-volumes.md

```diff
@@ -0,0 +1,55 @@
+---
+title: Persist data with .NET Aspire using volume mounts
+description: Learn about the .NET Aspire service defaults project.
+ms.date: 12/08/2023
+ms.topic: reference
+---
+
+# Persist .NET Aspire app data using volumes
+
+In this article, you learn how to configure .NET Aspire apps to persist data across app launches using volumes. A continuous set of data during local development is useful in many scenarios. Various .NET Aspire container types are able to leverage volume storage, such as PostreSQL, Redis and Azure Storage.
+
+## Understand volumes
+
+Volumes are the recommended way to persist data generated by containers and supported on both Windows and Linux. Volumes can store data from multiple containers at a time, offer high performance and are easy to back up or migrate. With .NET Aspire, you can configure a volume for each resource container using the <xref:Aspire.Hosting.ContainerResourceBuilderExtensions.WithVolumeMount%2A?displayProperty=nameWithType> method, which accepts three parameters:
+
+- **Source**: The source path of the volume, which is the physical location on the host.
+- **Target**: The target path in the container of the data you want to persist.
+- **VolumeMountType**: There are two types of volume mounts available:
+  - **Named mounts**: Storage managed by docker that your containers can use to persist data.
+  - **Bind mounts**: Files mounted from your host machine onto your container.
+
+Consider the following code:
+
+```csharp
+var sql = builder.AddSqlServerContainer("sql", sqlpassword)
+    .WithVolumeMount("VolumeMount.sqlserver.data", "/var/opt/mssql", VolumeMountType.Named)
+    .AddDatabase("sqldb");
+```
+
+In this example:
+
+- `VolumeMount.sqlserver.data` sets where the volume will be stored on the host.
+- `/var/opt/mssql` sets the path to the database files in the container.
+- `VolumeMountType.Named` sets a named volume that will persist across container life cycles.
+
+## Create a persistent password
+
+Storing data in a named volume requires a consistent password between app launches. Use the following command to set a local password in your .NET user secrets:
+
+```dotnetcli
+dotnet user-secrets set samplepassword <password>
+```
+
+You can retrieve this password in your app using the following code:
+
+```csharp
+// Retrieve the password you created from user secrets
+var password = builder.Configuration["samplepassword"];
+```
+
+## Persist data using volume mounts
+
+Add the following code to the _Program.cs_ file in your **AppHost** project to persist your desired data across app launches:
+
+:::code source="snippets/persist-data-volume-mounts/AspireSQLEFCore.AppHost/Program.cs":::
```

Comment:
I think we need a summary section, or something that recaps the benefits of this approach, and what problems it solves.